dev/bots/prepare_package.dart

```diff
@@ -426,6 +428,104 @@ class ArchiveCreator {
     await _unzipArchive(gitFile, workingDirectory: minGitPath);
   }
 
+  /// Downloads an archive of every package that is present in the temporary
+  /// pub-cache from pub.dev. Stores the archives in
+  /// $flutterRoot/.pub-preload-cache.
+  ///
+  /// These archives will be installed in the user-level cache on first
+  /// following flutter command that accesses the cache.
+  ///
+  /// Precondition: all packages currently in the PUB_CACHE of [_processRunner]
+  /// are installed from pub.dev.
+  Future<void> _downloadPubPackageArchives() async {
+    final Pool pool = Pool(10); // Number of simultaneous downloads.
+    final http.Client client = http.Client();
+    final Directory preloadCache = Directory(path.join(flutterRoot.path, '.pub-preload-cache'));
```

Comment:
Do we need to add this to the top-level repo .gitignore?

Reply:
Yes! good point.
Done